Purchases/Purchasing/ProductDetails.swift

```diff
@@ -0,0 +1,117 @@
+//
+//  Product.swift
+//  Product
+//
+//  Created by Andrés Boedo on 7/16/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+import StoreKit
+
+@objc(RCProductDetails) public class ProductDetails: NSObject {
+    public override func isEqual(_ object: Any?) -> Bool {
+        return self.productIdentifier == (object as? ProductDetails)?.productIdentifier
+    }
+
+    @objc public var localizedDescription: String { fatalError() }
+    @objc public var localizedTitle: String { fatalError() }
+    @objc public var price: Decimal { fatalError() }
+    @objc public var localizedPriceString: String { fatalError() }
+    @objc public var productIdentifier: String { fatalError() }
+    @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 8.0, *)
+    @objc public var isFamilyShareable: Bool { fatalError() }
+
+    @available(iOS 12.0, macCatalyst 13.0, tvOS 12.0, macOS 10.14, watchOS 6.2, *)
+    @objc public var subscriptionGroupIdentifier: String? { fatalError() }
+
+    // todo: it looks like StoreKit 2 doesn't have support for these?
+    //    YES if this product has content downloadable using SKDownload
+    //    var isDownloadable: Bool { get }
+    //
+    //    var downloadContentLengths: [NSNumber] { get }
+    //
+    //    // Version of the downloadable content
+    //    var contentVersion: String { get }
+    //
+    //    var downloadContentVersion: String { get }
+    //
+
+    // todo: add subscription period
+    //    @available(iOS 11.2, *)
+    //    var subscriptionPeriod: SKProductSubscriptionPeriod? { get }
+    //
+
+    // todo: add product discounts
+    //    @available(iOS 11.2, *)
+    //    var introductoryPrice: SKProductDiscount? { get }
+    //    //
+    //    @available(iOS 12.2, *)
+    //    var discounts: [SKProductDiscount] { get }
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+@objc(RCSK2ProductDetails) public class SK2ProductDetails: ProductDetails {
+
+    // todo: remove when this gets fixed.
+    // limiting to arm architecture since builds on beta 5 fail if other archs are included
+    #if arch(arm64)
+    init(sk2Product: StoreKit.Product) {
```

Comment:
thoughts on making a typealias SK2Product -> StoreKit.Product and SK1Product -> SKProduct? or is this really the only place we should need to distinguish moving forward?

Reply:
that's brilliant and I love it! will do

Reply:
I haven't forgotten about this one, but I'm going to do it in a follow-up PR since it affects a whole bunch of files

this is what I was thinking: 

```swift
/// TypeAlias to the Original In-App Purchase Framework's Product type, called SKProduct
public typealias SK1Product = SKProduct

/// TypeAlias to the New In-App Purchase Framework's Product type, called StoreKit.Product
@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
public typealias SK2Product = StoreKit.Product

```

an alternative is to do `SK2Product -> StoreKit.Product and SK1Product -> SKProduct` like you said. I figured that the distinction between new and legacy might make it more obvious at a glance, so we don't accidentally confuse them

Reply:
let me know what you think!

Reply:
update: ended up going with `SK2Product` and `SK1Product`, since the legacy vs new naming got confusing quickly, and would be inconsistent with stuff that directly references sk1 or sk2